src/Functional/Exceptions/InvalidArgumentException.php

```diff
@@ -191,9 +191,10 @@ public static function assertBoolean($value, $callee, $parameterPosition)
         if (!is_bool($value)) {
             throw new static(
                 sprintf(
-                    '%s() expects parameter %d to be boolean',
+                    '%s() expects parameter %d to be boolean, %s given',
                     $callee,
-                    $parameterPosition
+                    $parameterPosition,
+                    gettype($value)
```

Comment:
to work with objects, you want;
`is_object($value) ? get_class($value) : gettype($value);`

Reply:
Reworked to include class when invalid value is an object.